backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/work/impl/DeleteByQueryWork.java

```diff
@@ -60,7 +64,7 @@ public Builder(URLEncodedString indexName, JsonObject payload, ElasticsearchWork
 			super( ElasticsearchRequestSuccessAssessor.DEFAULT_INSTANCE );
 			this.indexName = indexName;
 			this.payload = payload;
-			this.refreshWorkBuilder = workFactory.refresh().index( indexName );
+			this.refreshWorkBuilder = workFactory.isRefreshSupported() ? workFactory.refresh().index( indexName ) : null;
```

Comment:
I wonder if we could have some "constant/stub" builder so we would not need to make those other null checks in this class

Reply:
We could, but it seems to me that it would be worse because now we'd have a builder class to maintain, even though its only goal is to produce nothing... ?

Anyway, this code is covered by tests, so I'm not particularly worried about the smelly `null` usage.